Assets/Scripts/NPCs/NPCManager.cs

```diff
@@ -55,4 +90,63 @@ public float GetDistanceWithPlayer()
     {
         return Vector3.Distance(Indestructibles.Player.transform.position, transform.position);
     }
+
+    public bool IsChasing()
+    {
+        return _stateMachine.CurrentState.GetType() == typeof(Chase);
+    }
+    public void StartPunching()
+    {
+        if (_canPunch && !Punching)
+        {
+            Punching = true;
+            Animator.Play("Punch", PunchLayerIndex, 0f);
+            Animator.SetLayerWeight(PunchLayerIndex,1.0f); 
+        }
+    }
+
+    public void StopPunching()
+    {
+        Punching = false;
+    }
+    public void OnPlayerHit()
+    {
+        _stateMachine.StartWandering();
+        StopPunching();
+
+        _canChase = false;
+        Invoke(nameof(ChaseCooldown), 3.0f);
+    }
+
+    private void ChaseCooldown()
+    {
+        _canChase = true;
+    }
+    private void PunchCooldown()
+    {
+        _canPunch = true;
+    }
+    /*
+     * If the player collides with a NPC that is not chasing him, start chase.
+     * For now, the collider is of type trigger because there is no navmesh.
+     */
+    void OnCollisionEnter(Collision collision)
+    {
+        if (collision.gameObject.CompareTag("Player"))
+        {
+            if (_canChase && !IsChasing())
+            {
+                if (bumpingSounds.Length > 0)
```

Comment:
Can you reduced the amount of nesting please

Reply:
Done